app/lib/import_export.rb

```diff
@@ -396,143 +849,80 @@ def format_phone_number(phone)
-        healthcare_personnel_facility_name: patient[:healthcare_personnel_facility_name] || '',
-        crew_on_passenger_or_cargo_flight: patient[:crew_on_passenger_or_cargo_flight] || false,
-        member_of_a_common_exposure_cohort: patient[:member_of_a_common_exposure_cohort] || false,
-        member_of_a_common_exposure_cohort_type: patient[:member_of_a_common_exposure_cohort_type] || '',
-        exposure_risk_assessment: patient[:exposure_risk_assessment] || '',
-        monitoring_plan: patient[:monitoring_plan] || '',
-        exposure_notes: patient[:exposure_notes] || '',
-        status: '',
-        symptom_onset: patient[:symptom_onset]&.strftime('%F') || '',
-        case_status: patient[:case_status] || '',
-        lab_1_type: '',
-        lab_1_specimen_collection: '',
-        lab_1_report: '',
-        lab_1_result: '',
-        lab_2_type: '',
-        lab_2_specimen_collection: '',
-        lab_2_report: '',
-        lab_2_result: '',
-        jurisdiction_path: '',
-        assigned_user: patient[:assigned_user] || '',
-        gender_identity: patient[:gender_identity] || '',
-        sexual_orientation: patient[:sexual_orientation] || ''
-      }
+  # Creates a list of csv files from exported data
+  def csv_export(config, exported_data, index)
+    files = []
+    CUSTOM_EXPORT_OPTIONS.each_key do |data_type|
+      next unless config.dig(:data, data_type, :checked).present?
+
+      package = CSV.generate(headers: true) do |csv|
+        fields = config[:data][data_type][:checked]
+        csv << config[:data][data_type][:headers] || fields.map { |field| ALL_FIELDS_NAMES[data_type][field] }
+        exported_data[data_type].each do |record|
+          csv << fields.map { |field| record[field] }
+        end
+      end
+      files << { filename: build_export_filename(config, data_type, index), content: Base64.encode64(package) }
+    end
+    files
+  end
+
+  # Creates a list of excel files from exported data
+  def xlsx_export(config, exported_data, index)
+    if config[:separate_files].present?
+      files = []
+      CUSTOM_EXPORT_OPTIONS.each_key do |data_type|
+        next unless config.dig(:data, data_type, :checked).present?
+
+        Axlsx::Package.new do |p|
+          p.workbook.add_worksheet(name: config[:data][data_type][:tab] || CUSTOM_EXPORT_OPTIONS[data_type][:label]) do |sheet|
+            fields = config[:data][data_type][:checked]
+            sheet.add_row(config[:data][data_type][:headers] || fields.map { |field| ALL_FIELDS_NAMES[data_type][field] })
+            exported_data[data_type].each do |record|
+              sheet.add_row(fields.map { |field| record[field] }, { types: Array.new(fields.length, :string) })
+            end
+          end
+          files << { filename: build_export_filename(config, data_type, index), content: Base64.encode64(p.to_stream.read) }
+        end
+      end
+      files
+    else
+      Axlsx::Package.new do |p|
```

Comment:
Can we extract the main chunk of code that is shared between the if and else here into a separate function since it's almost exact the same except for the `files << ...`

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/1f13143f8968feff3c7e9ada6139b47e5cd4ecff#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR906